app/controllers/my_facilities_controller.rb

```diff
@@ -36,39 +36,18 @@ def index
   end
 
   def bp_controlled
-    @facilities = filter_facilities
-
-    @data_for_facility = {}
-
-    @facilities.each do |facility|
-      @data_for_facility[facility.name] = Reports::RegionService.new(region: facility, period: @period).call
-    end
-
-    @facilities_by_size = @facilities.group_by { |facility| facility.facility_size }
+    facilities = filter_facilities
+    process_facility_stats(facilities, "controlled_patients")
```

Comment:
One overall piece of feedback: I think the new service is doing too much and too tightly coupled to the RegionsService.
My suggestion - change your new service to take the `facility_data` from RegionsService and just do the math on the data...let the calling code worry about how to retrieve that data.  So for the `bp_controlled` action for example:

```ruby
def bp_controlled
  @facilities = filter_facilities
  @data_for_facility = {}		
  @facilities.each do |facility|		
     @data_for_facility[facility.name] = Reports::RegionService.new(region: facility, period: @period).call		
  end
  @stats_by_size = FacilityStatsService.new(@data_for_facility)
end
```

This kind of approach would hopefully make it easier to rip out or change the underlying implementation of RegionsService, which is coming 🔜 with https://github.com/simpledotorg/simple-server/pull/2014.

Reply:
Will do.